### PR TITLE
[FLINK-6210] close RocksDB in ListViaMergeSpeedMiniBenchmark && ListV…

### PR DESCRIPTION
**Pull request by fanyon**
close RocksDB in ListViaMergeSpeedMiniBenchmark && ListViaRangeSpeedMiniBenchmark

### CODE REVIEW

File: flink-contrib/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/benchmark/ListViaMergeSpeedMiniBenchmark.java [hunk 1]
```diff
@@ -50,55 +50,59 @@ public static void main(String[] args) throws Exception {
 
 		final RocksDB rocksDB = RocksDB.open(options, rocksDir.getAbsolutePath());
 
-		final String key = "key";
-		final String value = "abcdefghijklmnopqrstuvwxyz0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ7890654321";
+		try {
```
Comment (shixiaogang) [inline flink-contrib/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/benchmark/ListViaMergeSpeedMiniBenchmark.java]: I think it's better to use try-with-resources here.

File: flink-contrib/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/benchmark/ListViaMergeSpeedMiniBenchmark.java [hunk 2]
```diff
@@ -50,55 +50,59 @@ public static void main(String[] args) throws Exception {
...
-		rocksDB.get(keyBytes, resultHolder);
-		final long endGet3 = System.nanoTime();
+			final long beginGet3 = System.nanoTime();
+			rocksDB.get(keyBytes, resultHolder);
+			final long endGet3 = System.nanoTime();
 
-		System.out.println("end get - duration: " + ((endGet3 - beginGet3) / 1_000_000) + " ms");
+			System.out.println("end get - duration: " + ((endGet3 - beginGet3) / 1_000_000) + " ms");
+		} finally {
+			rocksDB.close();
```
Comment (shixiaogang) [inline flink-contrib/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/benchmark/ListViaMergeSpeedMiniBenchmark.java]: Besides `RocksDB`, `Options` and`WriteOptions` should also be closed here.

File: flink-contrib/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/benchmark/ListViaRangeSpeedMiniBenchmark.java [hunk 3]
```diff
@@ -54,59 +54,62 @@ public static void main(String[] args) throws Exception {
...
-			else {
-				break;
+			final long endInsert = System.nanoTime();
+			System.out.println("end insert - duration: " + ((endInsert - beginInsert) / 1_000_000) + " ms");
+
+			final byte[] resultHolder = new byte[num * valueBytes.length];
+
+			final long beginGet = System.nanoTime();
+
+			final RocksIterator iterator = rocksDB.newIterator();
```
Comment (shixiaogang) [inline flink-contrib/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/benchmark/ListViaRangeSpeedMiniBenchmark.java]: The iterator should be closed once it's not used. So it's better to use try-with-resources here.

File: flink-contrib/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/benchmark/ListViaRangeSpeedMiniBenchmark.java [hunk 4]
```diff
@@ -54,59 +54,62 @@ public static void main(String[] args) throws Exception {
...
 			}
-		}
 
-		final long endGet = System.nanoTime();
+			final long endGet = System.nanoTime();
 
-		System.out.println("end get - duration: " + ((endGet - beginGet) / 1_000_000) + " ms");
+			System.out.println("end get - duration: " + ((endGet - beginGet) / 1_000_000) + " ms");
+		} finally {
+			rocksDB.close();
```
Comment (shixiaogang) [inline flink-contrib/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/benchmark/ListViaRangeSpeedMiniBenchmark.java]: As with the problem mentioned above, all RocksDB objects should be closed once they are not used.

File: flink-contrib/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/benchmark/ListViaMergeSpeedMiniBenchmark.java (hunk 2 above)
Comment (shixiaogang) [inline flink-contrib/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/benchmark/ListViaMergeSpeedMiniBenchmark.java]: I think it's better to delete the RocksDB's directory here.

**Comment by fanyon** [comment]
Hi @shixiaogang , thank you for your reviews. Close and clean all resources such as temp directories is a good idea, I'll fix them later in this PR

**Comment by fanyon** [comment]
Hi @shixiaogang , as we discussed before, I have closed all resources in latest commit, thank you for your comments.

**Comment by StephanEwen** [comment]
Thank you for the patch!
Taking this over and merging this...

**Pull request closed without merging**